iron-form.html

```diff
@@ -456,6 +457,20 @@
 
         this.fire('iron-form-reset');
       }, 1);
+    },
+
+    _isChildOfRegisteredParent: function(node) {
+      var parent = Polymer.dom(node).parentNode;
```

Comment:
Also, I don't see why there is a problem that it returns false for `this._customElements[0]` (or rather, for any of the immediate children of the form). You only want this method to return `true` iff you're an element that has registered to the form AND you're nested (so you're in the shadow root of a different node, which is also in `this._customElements`